tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/InstanceDiscoveryManagerTests.cs

```diff
@@ -47,6 +48,7 @@ public override void TestInitialize()
             };
 
             _harness = base.CreateTestHarness();
+            TestCommon.CreateServiceBundleWithCustomHttpManager(_harness.HttpManager, validateAuthority: true);
```

Comment:
I think you are still missing a unit test that should verify the behavior of the InstanceManager when validateAuthority is set to false.

Reply:
Example test that shows how to inject a custom authority: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/fe547af3f0fd0404f32ad38496467808f124d549/tests/Microsoft.Identity.Test.Unit.net45/CoreTests/InstanceTests/B2cAuthorityTests.cs#L67

Reply:
Added.